codegen/src/main/java/software/amazon/awssdk/codegen/poet/builder/BaseClientBuilderClass.java

```diff
@@ -637,6 +658,30 @@ private MethodSpec defaultTokenAuthSignerMethod() {
                          .build();
     }
 
+    private MethodSpec defaultAuthSchemesMethod() {
+        TypeName returns = ParameterizedTypeName.get(ClassName.get(Map.class), ClassName.get(String.class),
+                                                     ParameterizedTypeName.get(ClassName.get(AuthScheme.class),
+                                                                               WildcardTypeName.subtypeOf(Object.class)));
+
+        MethodSpec.Builder builder = MethodSpec.methodBuilder("defaultAuthSchemes")
+                                               .addModifiers(PRIVATE)
+                                               .returns(returns);
+
+        List<String> schemesMapPopulate = new ArrayList<>();
+        for (AuthType authType : authSchemeSpecUtils.serviceDefaultAuthTypes()) {
```

Comment:
noted in the tests below, but this should be using a union of all auth types across all operations. In Smithy model this is easy from the auth traits on the service shape, but in C2J seems like we have to do the union.

Reply:
Yes, I missed this, I will fix it in the upcoming review.